File: CareerCup/LinkedIn/BinarySearch.java
public class BinarySearch
{
	public static void main(String args[])
	{
		int ar[] = new int[args.length];
		for(int i =0;i<args.length;i++)
		{
			ar[i] = Integer.parseInt(args[i]);
		}
		System.out.println(binarySearch(ar,12));

	}
	public static int binarySearch(int ar[], int key)
	{
		int start=0;
		int end = ar.length-1;
		while(start<=end)
		{
			int mid = (start+end)/2;
			if(ar[mid]==key)
			{
				return mid;
			}		
			else if (ar[mid]>key)
			{
				end = mid-1;
			}
			else
			{
				start = mid+1;
			}
		}
		return -1;
	

	}

}

File: CareerCup/FB/binarySearch.py
def binary_search(ar,key):
	
	start =0
	end = len(ar)-1
	while start<=end:
		mid = start+end/2
		if ar[mid]==key:
			return mid
		else:
			if ar[mid]<key:
				start = mid+1
			else:
				end = mid-1
	
	return -1

key = input()
print binary_search(map(int,raw_input().split(' ')),key)
